### PR TITLE
Feature/admin page

### PR DESCRIPTION
**Pull request by Galbator1x**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -7,4 +7,8 @@ class User < ApplicationRecord
   validates :first_name, presence: true, length: { minimum: 2 }
   validates :last_name, presence: true, length: { minimum: 2 }
   validates :email, presence: true, uniqueness: true, format: { with: /.*@.*/ }
+
+  def name
```
Comment (pn-y) [inline app/models/user.rb]: Presentation logic should not be in model

File: app/controllers/admin/application_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Admin::ApplicationController < ApplicationController
+  include Concerns::AuthHelper
+
+  before_action :authenticate_user!, :authorize
+  helper_method :current_user
```
Comment (pn-y) [inline app/controllers/admin/application_controller.rb]: It seems it's a current admin

File: app/controllers/admin/application_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def authorize
+    render(file: File.join(Rails.root, 'public/403.html'), status: 403, layout: false) if (forbidden?)
```
Comment (devality) [inline app/controllers/admin/application_controller.rb]: Follow common ruby style, don't wrap if condition in brackets.

File: app/controllers/admin/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to admin_users_url
+    else
+      render :new_admin_user_url
+    end
```
Comment (devality) [inline app/controllers/admin/users_controller.rb]: Use `respond_with`.

File: app/models/user.rb [hunk 5]
```diff
@@ -7,4 +7,8 @@ class User < ApplicationRecord
   validates :first_name, presence: true, length: { minimum: 2 }
   validates :last_name, presence: true, length: { minimum: 2 }
   validates :email, presence: true, uniqueness: true, format: { with: /.*@.*/ }
```
Comment (devality) [inline app/models/user.rb]: Are you sure you are using proper regexp for email? Try to enter `@@@`.

File: app/controllers/admin/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to admin_users_url
+    else
+      respond_with @user
```
Comment (devality) [inline app/controllers/admin/users_controller.rb]: Read how to use `respond_with`, use it in all `create` and `update` actions.

**Pull request merged**